backend/corpora/common/entities/project.py

```diff
@@ -52,8 +56,57 @@ def create(
             links=cls._create_sub_objects(
                 links, DbProjectLink, add_columns=dict(project_id=primary_key, project_status=status, **kwargs)
             ),
+            **kwargs,
         )
 
         cls.db.session.add(new_db_object)
         cls.db.commit()
         return cls(new_db_object)
+
+    @classmethod
+    def get_project(cls, project_uuid):
+        """
+        Given the project_uuid, retrieve a live project.
+        :param project_uuid:
+        """
+        return cls.get((project_uuid, ProjectStatus.LIVE.name))
+
+    @classmethod
+    def list_projects_in_time_range(cls, *args, **kwargs):
+        return cls.list_in_time_range(*args, filters=[DbProject.status == ProjectStatus.LIVE.name], **kwargs)
+
+    @classmethod
+    def list_in_time_range(
+        cls, to_date: int = None, from_date: int = None, filters: list = None
+    ) -> typing.List[typing.Dict]:
+        """
+        Queries the database for projects that have been created within the specified time range.
+
+        :param to_date: If provided, only lists projects that were created before this date. Format of param is Unix
+        timestamp since the epoch in UTC timezone.
+        :param from_date: If provided, only lists projects that were created after this date. Format of param is Unix
+        timestamp since the epoch in UTC timezone.
+        :param filters: additional filters to apply to the query.
+        :return: The results is a list of flattened dictionaries containing the `list_entities`
+        """
+
+        filters = filters if filters else []
+        list_entities = [DbProject.created_at, DbProject.id]
+
+        def to_dict(db_object):
```

Comment:
Is this not redundant to the `to_dict` method that objects already inherit?

Reply:
When you use `with_entities()` in the query, SQLAlchemy returns those fields as a fancy tuple with a bunch of extra metadata attached. This parses the `list_entities` into a more intuitive dictionary.